public/app/features/dashboard-scene/scene/DashboardDatasourceBehaviour.tsx

```diff
@@ -58,6 +82,25 @@ export class DashboardDatasourceBehaviour extends SceneObjectBase<DashboardDatas
 
     return () => {
       this.prevRequestId = sourcePanelQueryRunner?.state.data?.request?.requestId;
+      if (libraryPanelSub) {
+        libraryPanelSub.unsubscribe();
+      }
     };
   }
+
+  private handleLibPanelStateUpdates(
+    newLibPanel: LibraryPanelBehaviorState,
+    oldLibPanel: LibraryPanelBehaviorState,
+    queryRunner: SceneQueryRunner,
+    sourcePanel: VizPanel
+  ) {
+    if (newLibPanel && newLibPanel?.isLoaded && newLibPanel !== oldLibPanel) {
+      const libPanelQueryRunner = getQueryRunnerFor(sourcePanel);
+
+      if (!(libPanelQueryRunner instanceof SceneQueryRunner)) {
+        throw new Error('Could not find SceneQueryRunner for panel');
+      }
+      queryRunner.runQueries();
+    }
+  }
```

Comment:
not sure I understand what this code does (what scenario it is for), it works when I am not including it

Reply:
This is needed for slow library panels. When a libPanel loads slowly, the panel with dashboardDS will not find any data and show nothing. 

We are not running queries on lib panel, we are only checking for it's existence (assuming that it has loaded the data), and then just run the dashboardDS panel queries to pull data from lib panel